src/argon2.c

```diff
@@ -445,8 +445,8 @@ const char *argon2_error_message(int error_code) {
 }
 
 size_t argon2_encodedlen(uint32_t t_cost, uint32_t m_cost, uint32_t parallelism,
-                         uint32_t saltlen, uint32_t hashlen, argon2_type type) {
+                         size_t saltlen, uint32_t hashlen, argon2_type type) {
   return strlen("$$v=$m=,t=,p=$$") + strlen(argon2_type2string(type, 0)) +
          numlen(t_cost) + numlen(m_cost) + numlen(parallelism) +
-         b64len(saltlen) + b64len(hashlen) + numlen(ARGON2_VERSION_NUMBER) + 1;
```

Comment:
Replacing ``b64len(saltlen)`` with ``saltlen`` is not good!
See ``size_t b64len(uint32_t len)`` 
https://github.com/P-H-C/phc-winner-argon2/blob/d3d81d3002c65e3a1738dd11fdb19bf8ba23349d/src/encoding.c#L440-L453

I think a clean solution might be to do as suggested by in  https://github.com/P-H-C/phc-winner-argon2/issues/201#issuecomment-328027862

@khovratovich, @Leont: would it be possible to check the second option and comment or create a PR?

Reply:
Could you provide a solution that works as you suggested?